habitat/tasks/rearrange/actions.py

```diff
@@ -77,25 +85,22 @@ def _grasp(self):
 
             closest_obj_pos = scene_obj_pos[closest_obj_idx]
             to_target = np.linalg.norm(ee_pos - closest_obj_pos, ord=2)
+            sim_idx = self._sim.scene_obj_ids[closest_obj_idx]
 
-            if to_target < self.thresh_dist:
-                self._sim.set_snapped_obj(closest_obj_idx)
+            if to_target < self._config.GRASP_THRESH_DIST:
+                self._sim.grasp_mgr.snap_to_obj(sim_idx)
 
-        # Get the marker the EE is closest to.
-        marker_name_pos = self._sim.get_marker_positions()
-        if self.snap_markers and len(marker_name_pos) > 0:
-            marker_name, marker_pos = zip(*marker_name_pos.items())
-            marker_pos = np.array(marker_pos)
-            closest_marker_idx = np.argmin(
-                np.linalg.norm(ee_pos - marker_pos, axis=-1)
-            )
-            closest_marker_pos = marker_pos[closest_marker_idx]
-            to_marker = np.linalg.norm(ee_pos - closest_marker_pos)
-            if to_marker < self.thresh_dist:
-                self._sim.set_snapped_marker(marker_name[closest_marker_idx])
+    def _ungrasp(self):
+        self._sim.desnap_object()
+
+    def step(self, grip_ac, should_step=True, **kwargs):
```

Comment:
To avoid breaking API:
```suggestion
    def step(self, *args, grip_ac, should_step=True, **kwargs):
```

Reply:
What `grip_ac` stands for? We need better name.

Reply:
`grip_ac` is for grip action, changed all shorthands for `ac` to `action`.